src/test/fuzz/coinscache_sim.cpp

```diff
@@ -151,14 +148,10 @@ class CoinsViewBottom final : public CCoinsView
     {
```

Comment:
I was surprise to see that the method called `GetCoin` now:
* overwrites the dummy coin parameter with the value found in the cache
* returns a boolean indicating whether the coin exists in the cache and is not spent (the name doesn't have any hints about it being spent or not)

I don't mind if you think it's outside the scope of the current change, but checking the usages it seems to me we can now simplify `GetCoin` to actually return the coin and let the callsite check for spent and it might simplify the usages as well:
```C++
std::optional<Coin> CCoinsViewCache::GetCoin(const COutPoint& outpoint) const
{
    auto it = FetchCoin(outpoint);
    if (it != cacheCoins.end()) {
        return it->second.coin;
    }
    return std::nullopt;
}
```

<details>
<summary>... and in tests</summary>
```C++
std::optional<Coin> GetCoin(const COutPoint& outpoint) const
{
    if (const auto [it, found] = m_data.find(outpoint); found) {
        return it->second;
    }
    return std::nullopt;
}
```
</details>

Which would simplify usages from e.g.
```C++
Coin coin_using_get_coin;
const bool exists_using_get_coin = coins_view_cache.GetCoin(random_out_point, coin_using_get_coin);
if (exists_using_get_coin) {
    assert(coin_using_get_coin == coin_using_access_coin);
}
assert((exists_using_access_coin && exists_using_have_coin_in_cache && exists_using_have_coin && exists_using_get_coin) ||
       (!exists_using_access_coin && !exists_using_have_coin_in_cache && !exists_using_have_coin && !exists_using_get_coin));
```
to something like:
```C++
if (auto coin_optional = coins_view_cache.GetCoin(random_out_point); coin_optional && !coin_optional->IsSpent()) {
    assert(*coin_optional == coin_using_access_coin);
    assert(exists_using_access_coin && exists_using_have_coin_in_cache && exists_using_have_coin);
} else {
    assert(!exists_using_access_coin && !exists_using_have_coin_in_cache && !exists_using_have_coin);
}
```

What do you think?

Reply:
This behavior is documented [here](https://github.com/bitcoin/bitcoin/blob/master/src/coins.h#L306-L310), so not sure why test code bothered to do this. Indeed, switching to `std::optional` would be cleaner, but I think a much bigger refactor. This is also suggested in the original PR [here](https://github.com/bitcoin/bitcoin/pull/18746#issuecomment-842393167).

Reply:
> but I think a much bigger refactor

Maybe, would it help if I did that in a different PR?

Reply:
Yes, but it would conflict heavily with this one. Many of the tests would have to be removed as well.

Reply:
I'm fine with that, extracted to a branch for now, will push as a draft after your PR stabilizes

Reply:
you can resolve this comment, I see now that this should definitely be investigated in a separate PR (it's a big change).

Reply:
I think you can resolve your own comment threads too 😉

Reply:
I don't have write access to yiur PR:
https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/reviewing-changes-in-pull-requests/about-pull-request-reviews#resolving-conversations